### ISSUE TITLE
Cognito User Pool: Add support for case sensitivity flag

### ISSUE THREAD

**Issue by bacoboy** [terraform-providers/terraform-provider-aws#12094]
Amazon [recently announced](https://forums.aws.amazon.com/ann.jspa?annID=7459) a setting on user pools in cognito to ignore case.  The default is case sensitive so:

```
Joe.Smith@example.com
```
is a different user than:
```
joe.smith@example.com
```

Turning this flag on says these are the same thing.

On the command line, I believe this maps to the `--username-configuration` flag which takes a new json blob:
```
{
  "CaseSensitive": true|false
}
```
or the shorthand:
```
CaseSensitive=boolean

```
[Reference](https://docs.aws.amazon.com/cli/latest/reference/cognito-idp/create-user-pool.html)

**Comment by bflad** [comment]
Support for an `username_configuration` configuration block in the `aws_cognito_user_pool` resource has been merged and will release with version 2.54.0 of the Terraform AWS Provider, later this week. Thanks to @garretruh for the implementation. 👍

**Comment by bacoboy** [comment]
Thanks @bflad for pushing this (and others) across the finish line! 🏁
